@dcpagotto looks like this PR breaks the build?  Can you see the errors, or do I need to copy them over?  Here is one:

```
FileSystem.cpp:(.text+0xe23): undefined reference to `Os::FileSystemInterface::FILE_SYSTEM_FILE_CHUNK_SIZE'
```

--- a/Fw/Types/BasicTypes.h
+++ b/Fw/Types/BasicTypes.h
@@ -88,8 +88,10 @@ typedef double F64;   //!< 64-bit floating point (double). Required for compiler
 /* Useful macro definitions                                                   */
 /*----------------------------------------------------------------------------*/
 #define FW_NUM_ARRAY_ELEMENTS(a) (sizeof(a) / sizeof((a)[0]))  //!< number of elements in an array
-#define FW_MAX(a, b) (((a) > (b)) ? (a) : (b))                 //!< MAX macro
-#define FW_MIN(a, b) (((a) < (b)) ? (a) : (b))                 //!< MIN macro
+// Deprecated: prefer std::max/std::min from <algorithm> in C++ code.
+// Kept for C compatibility only.
+#define FW_MAX(a, b) (((a) > (b)) ? (a) : (b))                 //!< MAX macro (deprecated in C++, use std::max)
+#define FW_MIN(a, b) (((a) < (b)) ? (a) : (b))                 //!< MIN macro (deprecated in C++, use std::min)
 
 #define FW_NO_ASSERT (1)  //!< Asserts turned off
 #define FW_FILEID_ASSERT \

--- a/Os/FileSystem.cpp
+++ b/Os/FileSystem.cpp
@@ -298,6 +298,7 @@ FileSystem::Status FileSystem::copyFileData(File& source, File& destination, FwS
     for (copiedSize = 0; (copiedSize < size) && (i < maximum); copiedSize += chunkSize, i++) {
         // chunkSize is FILE_SYSTEM_FILE_CHUNK_SIZE unless size-copiedSize is less than that
         // in which case chunkSize is size-copiedSize, ensuring the last chunk reads the remaining data
+        // static_cast needed to avoid ODR linker issue with static constexpr passed by reference to std::min
         chunkSize = std::min(static_cast<FwSizeType>(FILE_SYSTEM_FILE_CHUNK_SIZE), size - copiedSize);
         file_status = source.read(fileBuffer, chunkSize, Os::File::WaitType::WAIT);
         if (file_status != File::OP_OK) {

--- a/Os/FileSystem.cpp
+++ b/Os/FileSystem.cpp
@@ -298,7 +298,7 @@ FileSystem::Status FileSystem::copyFileData(File& source, File& destination, FwS
     for (copiedSize = 0; (copiedSize < size) && (i < maximum); copiedSize += chunkSize, i++) {
         // chunkSize is FILE_SYSTEM_FILE_CHUNK_SIZE unless size-copiedSize is less than that
         // in which case chunkSize is size-copiedSize, ensuring the last chunk reads the remaining data
-        chunkSize = std::min(FILE_SYSTEM_FILE_CHUNK_SIZE, size - copiedSize);
+        chunkSize = std::min(static_cast<FwSizeType>(FILE_SYSTEM_FILE_CHUNK_SIZE), size - copiedSize);
         file_status = source.read(fileBuffer, chunkSize, Os::File::WaitType::WAIT);
         if (file_status != File::OP_OK) {
             return FileSystem::handleFileError(file_status);

--- a/Os/Generic/PriorityQueue.cpp
+++ b/Os/Generic/PriorityQueue.cpp
@@ -200,7 +200,6 @@ QueueInterface::Status PriorityQueue::send(const U8* buffer,
         FW_ASSERT(this->m_handle.m_heap.push(priority, index));
         this->m_handle.store_data(index, buffer, size);
         this->m_handle.m_sizes[index] = size;
-        // track high water mark for queue depth monitoring
         this->m_handle.m_highMark = std::max(this->m_handle.m_highMark, this->getMessagesAvailable());
     }
     this->m_handle.m_empty.notify();

--- a/Os/Stub/test/ut/StubQueueTests.cpp
+++ b/Os/Stub/test/ut/StubQueueTests.cpp
@@ -38,10 +38,10 @@ TEST(Interface, Create) {
     Fw::String name = "My queue";
     const FwSizeType depth =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     const FwSizeType messageSize =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     Os::Stub::Queue::Test::StaticData::data.createStatus = Os::QueueInterface::Status::INVALID_PRIORITY;
     Os::QueueInterface::Status status = queue.create(0, name, depth, messageSize);
     ASSERT_EQ(Os::Stub::Queue::Test::StaticData::data.lastCalled, Os::Stub::Queue::Test::StaticData::CREATE_FN);
@@ -58,10 +58,10 @@ TEST(Interface, SendPointer) {
     Fw::String name = "My queue";
     const FwSizeType depth =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     const FwSizeType messageSize = sizeof Os::Stub::Queue::Test::InjectableStlQueueHandle::Message::data;
     const FwQueuePriorityType priority = STest::Random::lowerUpper(
-        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), std::numeric_limits<U32>::max()));
+        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), static_cast<FwQueuePriorityType>(std::numeric_limits<U32>::max())));
     U8 buffer[messageSize];
     ASSERT_EQ(Os::QueueInterface::Status::OP_OK, queue.create(0, name, depth, messageSize));
     ASSERT_STREQ(name.toChar(), queue.getName().toChar());
@@ -85,10 +85,10 @@ TEST(Interface, SendBuffer) {
     Fw::String name = "My queue";
     const FwSizeType depth =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     const FwSizeType messageSize = sizeof Os::Stub::Queue::Test::InjectableStlQueueHandle::Message::data;
     const FwQueuePriorityType priority = STest::Random::lowerUpper(
-        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), std::numeric_limits<U32>::max()));
+        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), static_cast<FwQueuePriorityType>(std::numeric_limits<U32>::max())));
     U8 storage[messageSize];
     Fw::ExternalSerializeBuffer buffer(storage, sizeof storage);
     Fw::String message = "hello";
@@ -115,10 +115,10 @@ TEST(Interface, ReceivePointer) {
     Fw::String name = "My queue";
     const FwSizeType depth =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     const FwSizeType sizeOut = sizeof Os::Stub::Queue::Test::InjectableStlQueueHandle::Message::data;
     const FwQueuePriorityType priorityOut = STest::Random::lowerUpper(
-        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), std::numeric_limits<U32>::max()));
+        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), static_cast<FwQueuePriorityType>(std::numeric_limits<U32>::max())));
 
     FwSizeType size = sizeof Os::Stub::Queue::Test::InjectableStlQueueHandle::Message::data;
     FwQueuePriorityType priority;
@@ -150,10 +150,10 @@ TEST(Interface, ReceiveBuffer) {
     Fw::String name = "My queue";
     const FwSizeType depth =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     const FwSizeType sizeOut = sizeof Os::Stub::Queue::Test::InjectableStlQueueHandle::Message::data;
     const FwQueuePriorityType priorityOut = STest::Random::lowerUpper(
-        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), std::numeric_limits<U32>::max()));
+        0, std::min(std::numeric_limits<FwQueuePriorityType>::max(), static_cast<FwQueuePriorityType>(std::numeric_limits<U32>::max())));
 
     FwSizeType size = sizeof Os::Stub::Queue::Test::InjectableStlQueueHandle::Message::data;
     FwQueuePriorityType priority;
@@ -186,7 +186,7 @@ TEST(Interface, MessageCount) {
     const FwSizeType messageSize = 1;
     const FwSizeType messages =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     ASSERT_EQ(Os::QueueInterface::Status::OP_OK, queue.create(0, name, depth, messageSize));
     Os::Stub::Queue::Test::StaticData::data.messages = messages;
     ASSERT_EQ(queue.getMessagesAvailable(), messages);
@@ -201,7 +201,7 @@ TEST(Interface, MessageHighWaterMarkCount) {
     const FwSizeType messageSize = 200;
     const FwSizeType highWater =
         STest::Random::lowerUpper(std::numeric_limits<FwSizeType>::min(),
-                                  std::min(std::numeric_limits<FwSizeType>::max(), std::numeric_limits<U32>::max()));
+                                  std::min(std::numeric_limits<FwSizeType>::max(), static_cast<FwSizeType>(std::numeric_limits<U32>::max())));
     ASSERT_EQ(Os::QueueInterface::Status::OP_OK, queue.create(0, name, depth, messageSize));
     Os::Stub::Queue::Test::StaticData::data.highWaterMark = highWater;
     ASSERT_EQ(queue.getMessageHighWaterMark(), highWater);

--- a/Utils/TokenBucket.cpp
+++ b/Utils/TokenBucket.cpp
@@ -73,7 +73,6 @@ bool TokenBucket ::trigger(const Fw::Time time) {
         Fw::Time nextTime = Fw::Time::add(this->m_time, replenishInterval);
         while (this->m_tokens < this->m_maxTokens && nextTime <= time) {
             // replenish by replenish rate, or up to maxTokens
-            // replenish tokens up to max capacity
             this->m_tokens += std::min(this->m_replenishRate, this->m_maxTokens - this->m_tokens);
             this->m_time = nextTime;
             nextTime = Fw::Time::add(this->m_time, replenishInterval);

